src/librustdoc/json/mod.rs

```diff
@@ -69,6 +69,20 @@ impl JsonRenderer<'tcx> {
                     .iter()
                     .filter_map(|i| {
                         let item = &i.impl_item;
+
+                        // HACK(hkmatsumoto): On impls for primitive types, we push the item
+                        // to the index even if it is not local. This is because we have to
+                        // index `std` when getting information of methods of primitive types
+                        // while many primitive types are originates from `core`, as a workaround
+                        // for the behavior of `ExternalCrate::primitives`.
+                        if let clean::types::ItemKind::ImplItem(ref impl_) = *item.kind {
+                            if impl_.trait_.is_none() {
+                                if let clean::types::Type::Primitive(_) = impl_.for_ {
+                                    self.item(item.clone()).unwrap();
+                                    return Some(from_item_id(item.def_id));
+                                }
+                            }
+                        }
```

Comment:
Can you make it more clear that this is only necessary because primitives can have impls from other crates, due to `lang_items`? Just move this whole block down into the `else` below.